### PR TITLE
Temporary project artificat infrastructure

### PR DESCRIPTION
**Pull request by ameukam**
Ref: https://github.com/kubernetes/k8s.io/issues/1758.

Create a GCP project allowing II Coop folks to prototype cross-cloud
artifact hosting infrastructure.

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

**Comment by ameukam** [comment]
/hold
Explicit approval from chairs and TLs.
/assign @spiffxp @thockin 
cc @dims

File: infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf
Comment (ameukam) [inline infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf:10]: @spiffxp will appear in the audit report. Created by `terraform init`.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [inline infra/gcp/clusters/projects/artifact-management-sandbox/main.tf:32]: Since this is temporary, I don't think we should the terraform module `gke-project`. Open to suggestions.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [inline infra/gcp/clusters/projects/artifact-management-sandbox/main.tf:43]: Supposed to be the same services as `k8s-artifacts-prod` : https://github.com/kubernetes/k8s.io/issues/1758#issuecomment-822685501. 
I would like the rest of the services added by hh's team.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [inline infra/gcp/clusters/projects/artifact-management-sandbox/main.tf:31]: Since this is temporary, I don't think we should use the terraform module `gke-project`. Open to suggestions.

File: infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf
Comment (ameukam) [inline infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf:11]: @spiffxp will appear in the audit report. Created by `terraform init`.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (hh) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf:43]: 👍

Comment (hh) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf:43]: Will update this once we merge / we explore current access.

**Comment by thockin** [comment]
No objections.  Leaving hold for Aaron JIC

/lgtm

File: infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf
Comment (spiffxp) [reply infra/gcp/clusters/projects/artifact-management-sandbox/provider.tf:11]: ack, I will try to capture this as part of https://github.com/kubernetes/k8s.io/pull/1952

File: infra/gcp/clusters/.gitignore
Comment (spiffxp) [inline infra/gcp/clusters/.gitignore:6]: This matches both, unless there's a need to allow `*.tfstatefoo`?
```suggestion
*.tfstate*
```

File: infra/gcp/clusters/.gitignore
Comment (spiffxp) [inline infra/gcp/clusters/.gitignore:13]: Unless there's a need to allow `foooverride.tfbar`?
```suggestion
*override.tf*
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
Some suggestions

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf]: Fixed.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf]: Adressed.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf]: @spiffxp Fixed.

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf]: Fixed.

File: infra/gcp/clusters/.gitignore
Comment (ameukam) [reply infra/gcp/clusters/.gitignore]: Fixed

File: infra/gcp/clusters/projects/artifact-management-sandbox/main.tf
Comment (ameukam) [reply infra/gcp/clusters/projects/artifact-management-sandbox/main.tf]: Fixed.

File: infra/gcp/clusters/.gitignore
Comment (ameukam) [reply infra/gcp/clusters/.gitignore]: Fixed

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by spiffxp — APPROVED** [review]
Thanks for addressing, remove hold when ready to deploy

**Pull request merged**
